operator/v1/types_insights.go

```diff
@@ -0,0 +1,134 @@
+	// all gatherers during the last gather event.
+	// +optional
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration,omitempty"`
+	// gatherers is a list of active gatherers (and their statuses) in the last gathering.
+	// +listType=atomic
+	// +optional
+	Gatherers []GathererStatus `json:"gatherers,omitempty"`
+}
+
+// insightsReport provides Insights health check report based on the most
+// recently sent Insights data.
+type InsightsReport struct {
+	// healthChecks provides basic information about active Insights health checks
+	// in a cluster.
+	// +listType=atomic
+	// +optional
+	HealthChecks []HealthCheck `json:"healthChecks,omitempty"`
+}
+
+// healthCheck represents an Insights health check attributes.
+type HealthCheck struct {
+	// description provides basic description of the healtcheck.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=2048
+	// +kubebuilder:validation:MinLength=10
+	Description string `json:"description"`
+	// totalRisk of the healthcheck. Indicator of the total risk posed
+	// by the detected issue; combination of impact and likelihood. The values can be from 1 to 4,
+	// and the higher the number, the more important the issue.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=4
+	TotalRisk int32 `json:"totalRisk"`
+	// advisorURI provides the URL link to the Insights Advisor.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https:\/\/\S+`
+	AdvisorURI string `json:"advisorURI"`
+	// state determines what the current state of the health check is.
+	// Health check is enabled by default and can be disabled
+	// by the user in the Insights advisor user interface.
+	// +kubebuilder:validation:Required
+	State HealthCheckState `json:"state"`
+}
+
+// healthCheckState provides information about the status of the
+// health check (for example, the health check may be marked as disabled by the user).
+// +kubebuilder:validation:Enum:=Enabled;Disabled
+type HealthCheckState string
+
+const (
+	// enabled marks the health check as enabled
+	HealthCheckEnabled HealthCheckState = "Enabled"
+	// disabled marks the health check as disabled
+	HealthCheckDisabled HealthCheckState = "Disabled"
+)
+
+// gathererStatus represents information about a particular
+// data gatherer.
+type GathererStatus struct {
+	// conditions provide details on the status of each gatherer.
+	// +listType=atomic
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Conditions []metav1.Condition `json:"conditions"`
+	// name is the name of the gatherer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:MinLength=5
+	Name string `json:"name"`
+	// lastGatherDuration represents the time spent gathering.
+	// +kubebuilder:validation:Required
+	LastGatherDuration metav1.Duration `json:"lastGatherDuration"`
```

Comment:
Validation here too, note no zero this time as its required
```suggestion
	// +kubebuilder:validation:Pattern="^([0-9]+(\\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$"
	LastGatherDuration metav1.Duration `json:"lastGatherDuration"`
```